framework/web/AssetManager.php

```diff
@@ -576,6 +592,10 @@ public function getPublishedUrl($path)
      */
     protected function hash($path)
     {
+        if (!empty($this->hashCallback)) {
```

Comment:
@klimov-paul , @samdark 

This should be `$this->hashCallback !== null` since `empty()` make sense with arrays and in some very rare situations for other types. Whole framework at least parts which written by @qiangxue used this convention. Why on review stage other core developers did not see such things?